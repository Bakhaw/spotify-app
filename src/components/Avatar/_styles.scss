.Avatar {
  background: #191414;

  &--default {
    border-radius: 500px;
    height: 150px;
    width: 150px;
  }

  &--small {
    border-radius: 10px;
    height: 36px;
    width: 36px;
  }

  &--no-image {
    background: #191414;
    border-radius: 10px;
    height: 36px;
    width: 36px;
  }
}
