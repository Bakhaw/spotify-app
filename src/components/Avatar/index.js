import React from 'react';

function Avatar({ artist }) {
  const isImageNull = artist.image === null;

  return isImageNull ? (
    <div className='Avatar--no-image' />
  ) : (
    <img alt={artist.name} className='Avatar' src={artist.image} />
  );
}

export default Avatar;
