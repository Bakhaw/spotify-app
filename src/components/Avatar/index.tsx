const Avatar: React.FC<React.ImgHTMLAttributes<HTMLImageElement>> = ({
  ...props
}) => {
  return <img className='Avatar' {...props} />;
};
export default Avatar;
