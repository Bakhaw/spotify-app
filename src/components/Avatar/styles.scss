.Avatar {
    border-radius: 500px;
    width: 150px;
    box-shadow: 1px 4px 10px 2px rgba(0, 0, 0, 0.25);
}