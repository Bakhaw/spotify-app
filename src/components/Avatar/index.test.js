import React from 'react'
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Avatar from '.';

test('components/Avatar/', () => {
  const artist = {
    image: 'damso.jpg',
  };

  const { container } = render(<Avatar artist={artist} />);
  expect(container.firstChild).toMatchSnapshot();
})