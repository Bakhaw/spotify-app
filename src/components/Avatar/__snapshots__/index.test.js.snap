// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Avatar/ 1`] = `
<div>
  <img
    class="Avatar"
    src="damso.jpg"
  />
</div>
`;
